### PR TITLE
net-p2p/rtorrent: add 0.10.0 & net-libs/libtorrent: add 0.14.0

### PR DESCRIPTION
**Pull request by PPN-SD**
**net-p2p/rtorrent: add 0.10.0**
fix SRC_URI
version required for curl removed

**net-libs/libtorrent: add 0.14.0**
fix SRC_URI
updated patch for zlib
new patch for remove tests for ip addresses
I think the workaround for instrumentation is no more required, but I can't test it.

---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: net-p2p/rtorrent/files/rtorrent-0.10.0-lto.patch [hunk 1]
```diff
@@ -0,0 +1,89 @@
+https://bugs.gentoo.org/861848
+https://github.com/rakshasa/rtorrent/issues/1264
+https://github.com/rakshasa/rtorrent/pull/1291
+https://github.com/rakshasa/rtorrent/commit/c1d076f1822752d4956a884a9aa9e6b0026b11f3
+https://github.com/rakshasa/rtorrent/commit/38b39bdafc9edd7e7e72672e6fcbd397b6c2cab8
+
+From c1d076f1822752d4956a884a9aa9e6b0026b11f3 Mon Sep 17 00:00:00 2001
+From: stickz <stickman002@mail.com>
+Date: Sun, 29 Sep 2024 13:24:01 -0400
+Subject: [PATCH] Fixing static building
```
Comment (eli-schwartz) [inline net-p2p/rtorrent/files/rtorrent-0.10.0-lto.patch:10]: This patch has nothing to do with LTO. It is a totally unrelated issue and should not have metadata in the patch header (or filename) talking about LTO.

You may link to https://github.com/rakshasa/rtorrent/pull/1296 instead...

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: net-p2p/rtorrent/files/rtorrent-0.10.0-lto.patch [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ 
+ AX_CXX_COMPILE_STDCXX([14], [noext], [mandatory])
+ 
+-- 
+2.45.2
+
+From 38b39bdafc9edd7e7e72672e6fcbd397b6c2cab8 Mon Sep 17 00:00:00 2001
+From: Adam Sampson <ats@offog.org>
+Date: Sun, 29 Sep 2024 22:14:16 +0100
+Subject: [PATCH] Add ax_require_defined.m4 to scripts.
```
Comment (eli-schwartz) [inline net-p2p/rtorrent/files/rtorrent-0.10.0-lto.patch:35]: This is just a vendored copy of dev-build/autoconf-archive anyway. It is used when running `eautoreconf`, which we admittedly do.

File: net-p2p/rtorrent/files/rtorrent-0.10.0-lto.patch (hunk 1 above)
Comment (PPN-SD) [reply net-p2p/rtorrent/files/rtorrent-0.10.0-lto.patch:10]: OK, I see. Thank for the clarification.

Comment (eli-schwartz) [reply net-p2p/rtorrent/files/rtorrent-0.10.0-lto.patch:10]: I am not sure what exactly the patch actually does, to be quite honest.

It causes /usr/bin/rtorrent to be compiled using libtool instead of `x86_64-pc-linux-gnu-g++` directly:'


```
# without
x86_64-pc-linux-gnu-g++ -std=c++14  -pipe -march=native -fstack-protector-all -O2 -fdiagnostics-color=always -frecord-gcc-switches -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing  -Wformat -Werror=format-security -DNDEBUG -Wall -isysroot / -pthread     -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -I/usr/include/ncursesw  -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,--sysroot,/ -o rtorrent main.o libsub_root.a -lpthread -lpthread  -lncursesw -ltinfow -lcppunit  -lcurl -ltorrent 


# with
libtool: link: x86_64-pc-linux-gnu-g++ -std=c++14 -pipe -march=native -fstack-protector-all -O2 -fdiagnostics-color=always -frecord-gcc-switches -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing -Wformat -Werror=format-security -DNDEBUG -Wall -isysroot / -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -I/usr/include/ncursesw -Wl,-O1 -Wl,-z -Wl,pack-relative-relocs -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,--sysroot -Wl,/ -o rtorrent main.o  -Wl,--as-needed libsub_root.a -lpthread -lncursesw -ltinfow -lcppunit -lcurl -ltorrent -pthread
```

libtool should not, in principle, be necessary for statically linking at all.

Comment (PPN-SD) [reply net-p2p/rtorrent/files/rtorrent-0.10.0-lto.patch:10]: Ok. I remove this **non**-lto.

Comment (PPN-SD) [reply net-p2p/rtorrent/files/rtorrent-0.10.0-lto.patch]: Sure.

**Review by eli-schwartz — COMMENTED** [review]
> new patch for zlib

I would reword this as an "updated" patch, since it's just rebasing files/libtorrent-0.13.8-sysroot.patch to files/libtorrent-0.14.0-sysroot.patch

File: net-libs/libtorrent/files/libtorrent-0.14.0-tests-address.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/test/Makefile.am b/test/Makefile.am
+index 0221796..6e23502 100644
+--- a/test/Makefile.am
++++ b/test/Makefile.am
+@@ -39,8 +39,6 @@ LibTorrent_Test_Common = \
+ 	helpers/utils.h
+ 
+ LibTorrent_Test_Torrent_Net_SOURCES = $(LibTorrent_Test_Common) \
+-	torrent/net/test_address_info.cc \
+-	torrent/net/test_address_info.h \
```
Comment (eli-schwartz) [inline net-libs/libtorrent/files/libtorrent-0.14.0-tests-address.patch:10]: Is there a bug somewhere that explains why you want to remove this?

Comment (PPN-SD) [reply net-libs/libtorrent/files/libtorrent-0.14.0-tests-address.patch:10]: This [line](https://github.com/rakshasa/libtorrent/blob/v0.14.0/test/torrent/net/test_address_info.cc#L25) causes a failure :  
```
torrent/net::test_address_info:
torrent/net::test_address_info::test_basic
ai_ref_err got wrong error, expected 'Name or service not known', got 'Temporary failure in name resolution'
 : assertion

[...]

- Expression: test_valid_ai_ref_err(std::bind(torrent::ai_get_addrinfo, "1.1.1.
300", nullptr, nullptr, std::placeholders::_1), EAI_NONAME)

Failures !!!
```

This could be resolved, but I think these tests are beyond the scope of the package.

File: net-libs/libtorrent/libtorrent-0.14.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,62 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools toolchain-funcs
+
+DESCRIPTION="BitTorrent library written in C++ for *nix"
+HOMEPAGE="https://rakshasa.github.io/rtorrent/"
+SRC_URI="https://github.com/rakshasa/rtorrent-archive/raw/master/${P}.tar.gz"
```
Comment (perfinion) [inline net-libs/libtorrent/libtorrent-0.14.0.ebuild:10]: whoa this is weird. It looks like the dev now has that repo with archives in it, weird. The hash of that file does match exactly with `https://github.com/rakshasa/rtorrent/releases/download/v0.10.0/libtorrent-0.14.0.tar.gz` 
the problem with that is the version of the release is the rtorrent version not the libtorrent one :(. I guess keeping the rtorrent-archive link is fine, im going to add a comment tho i think in case future people dont think its weird too

File: net-libs/libtorrent/files/libtorrent-0.14.0-tests-address.patch (hunk 3 above)
Comment (perfinion) [reply net-libs/libtorrent/files/libtorrent-0.14.0-tests-address.patch:10]: sounds like the network sandbox is blocking it. i agree that i'd rather drop skip the tests than enable the network tho yeah.

**Comment by perfinion** [comment]
Thanks for the PR, merged :)

File: net-libs/libtorrent/libtorrent-0.14.0.ebuild (hunk 4 above)
Comment (PPN-SD) [reply net-libs/libtorrent/libtorrent-0.14.0.ebuild:10]: Oh, I didn't pay attention. tags from github instead ?
https://github.com/rakshasa/libtorrent/archive/refs/tags/v0.14.0.zip

"https://github.com/rakshasa/${PN}/archive/refs/tags/v${PV}.zip"

I didn't check diffs for now.
